doc: add minutes for meeting 17 July 2024 (#1597)
nodejs/TSC: meetings/2024-07-17.md

# Node.js Technical Steering Committee (TSC) Meeting 2024-07-17

## Links

* **Recording**:  <https://www.youtube.com/watch?v=N4uIFgvRRUM>
* **GitHub Issue**: <https://github.com/nodejs/TSC/issues/1596>

## Present

* Antoine du Hamel @aduh95 (voting member)
* Benjamin Gruenbaum @benjamingr (voting member)
* Ruben Bridgewater @BridgeAR (voting member)
* Geoffrey Booth @GeoffreyBooth (voting member)
* Marco Ippolito @marco-ippolito (voting member)
* Matteo Collina @mcollina (voting member)
* Michael Dawson @mhdawson (voting member)
* Rafael Gonzaga @RafaelGSS (voting member)
* Richard Lau @richardlau (voting member)
* Robert Nagy @ronag (voting member)
* Paolo Insogna @ShogunPanda (voting member)
* Joe Sepi @<sepi@joesepi.com> (Guest - Node.js CPC rep)

## Agenda

### Announcements

* Rafael about to promote 22.5.0 !

### Reminders

> * Remember to nominate people for the [contributor spotlight](https://github.com/nodejs/node/blob/main/doc/contributing/reconizing-contributors.md#bi-monthly-contributor-spotlight)

Reviewer: Link has now moved: https://github.com/nodejs/node/pull/53990
```suggestion
* Remember to nominate people for the [contributor spotlight](https://github.com/nodejs/node/blob/main/doc/contributing/recognizing-contributors.md#bi-monthly-contributor-spotlight)
```

    chance that it might cause unforeseen issues. So better to have in 22 before goes LTS.
  * Geoffrey/Matteo correct.
  * No objections from TSC members in the meeting.

* inspector: add initial support for network inspection [#53593](https://github.com/nodejs/node/pull/53593)
  * Chrome helped us resolve the issue and Chengzhong is helping it along :)
  * Chengzhong: reaching out to Danil to draft a design doc
  * Benjamin, positive reaction from Chrome dev tools team and progress is being made.
  * agenda tag already removed

* module: add --experimental-strip-types [53725](https://github.com/nodejs/node/pull/53725)
  * Benjamin, strategic decisions, lots of stake holders. Good to make a few decisions to help
    conversation along.
    * Benjamin, ok to land and iterate, but we allow as much breakage etc.
    * Matteo, 2 takes.
      * First one is that we should try to have meeting with the Typescript team. Should wait until
        we get their feedback. Seems like the asked for some time. Ok with landing and removing
        later
    * Marco
      * Currently blocked as swc having trouble with some architectures so still some time before
         it would land anyway.

>       * Feedback from TypeScript team is very favorable. Created package to wrap swc which

Reviewer 2: ```suggestion
      * Feedback from TypeScript team is that there are some concerns that can be solved. Created package to wrap swc which
```

  * Benjamin, lets try and get Daniel on the TSC meeting next week and voice their concerns
  * Marco, have gotten some positive feedback from people on tc39
    * Proposal is to:
      * externalize into npm module
      * include the npm module in Node.js just like npm
      * already created the PR to do that.
      * will create WASM from source code
      * PR will look similar, but will include source code and wrap swc
  * Benjamin
    * Can we agree on landing
  * Antoine, only the compile time option is on the table because of the architectures side of
    Things. Marco believe latest version from swc will be addressed
  * Geoffrey, land but hold back from release, but increases the chances of backports
  * Richard, likely not a Node.js 22.x release in next few weeks anyway.
  * Michael
    * one approach, land in main with compile time option, backport that to 22.x, then remove
      the compile time option in main. That keeps most code consistent for backporting while
      not enabling in 22.x in shipping versions. If its not ready for 23 we could then re-add
      the compile time guard (or not if it is ready)

>     

Reviewer 3: ```suggestion
```
